packages/mui-utils/src/useOnMount.ts

```diff
@@ -0,0 +1,13 @@
+'use client';
+import * as React from 'react';
+
+const EMPTY = [] as unknown[];
+
+/**
+ * A React.useEffect equivalent that runs once, when the component is mounted.
+ */
+export default function useOnMount(fn: React.EffectCallback) {
```

Comment:
Is this really needed?

Reply:
In the datagrid codebase it's being used in a few different places like [here](https://github.com/mui/mui-x/blob/5ec566d6a3c8d8fb1e5f565eb83befba348af846/packages/grid/x-data-grid-pro/src/hooks/features/columnResize/useGridColumnResize.tsx#L719). We've added it because it make the semantic of the hook apparent, whereas `useEffect` is pretty generic. The dependency list being at the end also means it's harder to understand when an effect runs, e.g.:

```js
useEffect(() => {
  // lots of lines
}, [])
```

In this example, you don't know until the end of the effect callback that it's an effect that runs on mount only. Being able to read `useOnMount` right away makes reading the effect easier, because it's obvious when it runs. Avoiding the `[]` allocation & noise is also a nice touch.

Reply:
We might be moving all our hooks here or to a new package shortly: https://github.com/mui/mui-x/issues/11576#issuecomment-1893647652